src/sd/spi.rs

```diff
@@ -1,32 +1,23 @@
 use crate::sys::*;
 
-use core::{marker::PhantomData, ops::Deref};
+use core::{borrow::BorrowMut, marker::PhantomData, ops::Deref};
 
 use esp_idf_hal::{
     gpio::{InputPin, OutputPin},
     peripheral::Peripheral,
-    spi::SpiDriver,
+    spi::{SpiDeviceDriver, SpiDriver},
 };
 
 pub struct SpiDevice<'d> {
     configuration: sdspi_device_config_t,
-    driver: SpiDriver<'d>,
+    host: spi_host_device_t,
     _p: PhantomData<&'d mut ()>,
 }
 
-impl Drop for SpiDevice<'_> {
-    fn drop(&mut self) {
-        let result = unsafe { spi_bus_free(self.get_host()) };
-
-        if result != ESP_OK {
-            panic!("Failed to remove SPI device");
-        }
-    }
-}
-
 impl<'d> SpiDevice<'d> {
-    pub fn new(
-        driver: SpiDriver<'d>,
+    pub fn new<T>(
+        host: spi_host_device_t,
+        _: &mut SpiDeviceDriver<'d, T>,
```

Comment:
This is also incorrect at least for two reasons:
- You don't need the `SpiDeviceDriver` and you are not using it actually?!
- You are not holding onto the `SpiDeviceDriver` throughout the lifetime of `SpiDevice`?